src/app_service/service/transaction/service.nim

```diff
@@ -230,105 +230,150 @@ QtObject:
         self.txCounter[address] = networkChains
         self.threadpool.start(arg)
 
-  proc transfer*(
+  proc createApprovalPath*(self: Service, route: TransactionPathDto, from_addr: string, toAddress: Address, gasFees: string): TransactionBridgeDto =
+    var txData = TransactionDataDto()
+    let approve = Approve(
+      to: parseAddress(route.approvalContractAddress),
+      value: route.approvalAmountRequired,
+    )
+    let data = ERC20_procS.toTable["approve"].encodeAbi(approve)
+    txData = ens_utils.buildTokenTransaction(
+      parseAddress(from_addr),
+      toAddress,
+      $route.gasAmount,
+      gasFees,
+      route.gasFees.eip1559Enabled,
+      $route.gasFees.maxPriorityFeePerGas,
+      $route.gasFees.maxFeePerGasM
+    )
+    txData.data = data
+
+    var path = TransactionBridgeDto(bridgeName: "Simple", chainID: route.fromNetwork.chainId)
```

Comment:
Those names should be put in constants IMO

Reply:
done!